packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -116,6 +117,36 @@ export class PluginTree extends TreeImpl {
         return this._isEmpty;
     }
 
+    async refresh(raw?: CompositeTreeNode): Promise<CompositeTreeNode | undefined> {
+        const parent = !raw ? this._root : this.validateNode(raw);
+        let result: CompositeTreeNode | undefined;
+        if (CompositeTreeNode.is(parent)) {
+            const busySource = new CancellationTokenSource();
+            this.doMarkAsBusy(parent, 800, busySource.token);
+            try {
+                result = parent;
+                // Resolve all tree items to enable problem decorator
+                const children = await this.resolveAllChildren(parent);
```

Comment:
Do we want to make this a standard feature for all contributed tree views? From the behavior of some tree views in VSCode (e.g. in the GitHub plugin, the Reference View), it certainly seems as though resolving children and grand children is delayed until requested, and automatically resolving all children could come with substantial performance consequences for trees that rely on network calls to retrieve information.

Reply:
Yes, you are right. I tried to fetch all children to apply decorations for parent elements. In my case (in my extension) I send whole tree in one request and this change won't affect performance, but for other cases it won't be effective. I'll try to figure out how to fix it, thanks!